### DISCUSSION TITLE
Elm327 wifi problème avec mac m1

### DISCUSSION THREAD

**Discussion by enesokyay** [cedricp/ddt4all#1214] [Q&A]
Issue: Connection to ELM327 Wi-Fi Device Fails - Stuck on “Connecting to port 35000”

Description:

I am trying to connect DDT4All to my ELM327 Wi-Fi device on my MacBook running macOS. The device is connected to the Wi-Fi network, and I can confirm that the port 35000 is open and accessible (as seen in the attached terminal output). However, when I run DDT4All, it repeatedly tries to connect to the port but fails to establish a connection.

Steps Taken:

	1.	Wi-Fi Settings:
	•	Manually set the IP address to 192.168.0.2.
	•	Subnet mask set to 255.255.255.0.
	•	Router (Gateway) set to 192.168.0.10.
	•	Disabled IPv6.
(Please see the attached screenshot for my Wi-Fi configuration.)
	2.	Port Check:
	•	Using the nc -zv 192.168.0.10 35000 command, I confirmed that the port 35000 on the ELM327 device is open and reachable.
	3.	DDT4All Configuration:
	•	I verified that the IP and port in the DDT4All source code are correctly set to 192.168.0.10:35000.
(Please refer to the attached screenshots of the source code.)
	4.	Issue:
	•	Despite all these settings, DDT4All continues to try to connect to the ELM327 device, but the connection is never established. The software remains stuck on “Connecting to port 35000,” as shown in the terminal output screenshot.

Screenshots:

	•	Wi-Fi Configuration: [Insert attached screenshot of Wi-Fi settings]
	•	Port Check: [Insert attached screenshot of terminal output showing successful port check]
	•	DDT4All Terminal Output: [Insert attached screenshot of terminal output showing repeated connection attempts]
	•	Source Code: [Insert attached screenshot of source code showing IP and port settings]

Additional Information:

	•	The ELM327 device works fine when connected to a mobile device (phone) using the same Wi-Fi settings, so the issue seems to be specific to the MacBook or the DDT4All configuration on macOS.

Request:

Any guidance on what might be causing this issue or how to troubleshoot further would be greatly appreciated. Is there something specific to macOS or the DDT4All setup that could be causing this connection failure?

![IMG_D9E76581-2BB6-4845-A6C5-CC3584CECEDA](https://github.com/user-attachments/assets/f2b6702a-b02d-4461-8194-d22f5ef5ce0b)
![IMG_FFAB4FE3-287D-49A1-95B0-CA393B24F569](https://github.com/user-attachments/assets/43883c55-e577-469b-88f7-d793df98d5ba)
![IMG_4DA9564C-4B6A-4528-B0E8-401D6C35DE4A](https://github.com/user-attachments/assets/4382542c-9e5a-470d-82b1-68b3fcf81706)
![IMG_1573](https://github.com/user-attachments/assets/89ab5a1d-0988-4985-8d0d-6a4506e99ced)

**Comment by Furtif** [comment]
as tu tester sa vraiment connecté dans le véhicule ?

**Comment by enesokyay** [comment]
Yesterday I made a connection with my phone and I was able to make a change, but the application on the phone is quite limited. I can enter from my iPhone 12 device, both hand and the car work, but . I can't connect when I switch to the Mac

**Comment by enesokyay** [comment]
![image](https://github.com/user-attachments/assets/0d0c8f56-a684-4a59-b358-f342b2c15f1b)

I come across this when I try this way.

**Comment by Furtif** [comment]
je crois que tas pas laisser le temps a github pour upload l'image...
https://github.com/cedricp/ddt4all/issues/1213#issuecomment-2308451041
stressé probablement...

**Comment by enesokyay** [comment]
> je crois que tas pas laisser le temps a github pour upload l'image...
> 
> https://github.com/cedricp/ddt4all/issues/1213#issuecomment-2308451041
> 
> stressé probablement...

🫡🫡

**Comment by enesokyay** [comment]
> je crois que tas pas laisser le temps a github pour upload l'image...
> 
> https://github.com/cedricp/ddt4all/issues/1213#issuecomment-2308451041
> 
> stressé probablement...

I've been trying since last night I still haven't solved the problem. I even installed a Windows for Mac...

**Comment by Furtif** [comment]
> > je crois que tas pas laisser le temps a github pour upload l'image...
> > [#1213 (comment)](https://github.com/cedricp/ddt4all/issues/1213#issuecomment-2308451041)
> > stressé probablement...
> 
> I've been trying since last night I still haven't solved the problem. I even installed a Windows for Mac...

worked ??

**Comment by enesokyay** [comment]
> > > je crois que tas pas laisser le temps a github pour upload l'image...
> 
> > > [#1213 (comment)](https://github.com/cedricp/ddt4all/issues/1213#issuecomment-2308451041)
> 
> > > stressé probablement...
> 
> > 
> 
> > I've been trying since last night I still haven't solved the problem. I even installed a Windows for Mac...
> 
> 
> 
> worked ??

Non 😑![image](https://github.com/user-attachments/assets/3c0c5134-8b14-4af6-a816-31a2a5f9d1b1)![image](https://github.com/user-attachments/assets/2acb04f3-93fb-4a96-a949-be4d9a943879)![image](https://github.com/user-attachments/assets/2366a2b7-0f24-4159-b83a-7a4abb71a50c)

**Comment by enesokyay** [comment]
I can enter my obd device with my iPhone phone.![image](https://github.com/user-attachments/assets/c385fd6f-aae5-49a8-9a23-56afef7ff4e3)

**Comment by Furtif** [comment]
Bizarre tout sa, j’ai aussi se app dans mon iPhone mais mon obd2 Wi-Fi fonctionne grave bien avec tout ceci.

**Comment by Furtif** [comment]
Vous aviez cliquer sur l’icone Wi-Fi dans ddt4all ?

**Comment by enesokyay** [comment]
> Vous aviez cliquer sur l’icone Wi-Fi dans ddt4all ?

Oui

**Comment by enesokyay** [comment]
> Bizarre tout sa, j’ai aussi se app dans mon iPhone mais mon obd2 Wi-Fi fonctionne grave bien avec tout ceci.
> 
> 
> 
> 

Il n'y a pas de problème avec mon téléphone. Mais je ne peux pas me connecter avec mac. Je pense que je vais devoir acheter un obd connecté par USB...

**Comment by Furtif** [comment]
> > Bizarre tout sa, j’ai aussi se app dans mon iPhone mais mon obd2 Wi-Fi fonctionne grave bien avec tout ceci.
> 
> Il n'y a pas de problème avec mon téléphone. Mais je ne peux pas me connecter avec mac. Je pense que je vais devoir acheter un obd connecté par USB...

peut être désactivant le firewall, mais bon j'ai pas eu besoin de faire sa dans mon win11 ou mon macos ou linux debian.

**Comment by enesokyay** [comment]
> peut être désactivant le firewall, mais bon j'ai pas eu besoin de faire sa dans mon win11 ou mon macos ou linux debian.

Cela m'est venu à l'esprit et je l'ai fait, mais le résultat n'a pas changé.

**Reply by Furtif** [reply]
Chelou tout sa … mais bon là plus part wifi Bluetooth est ok … ceci concerne juste peut-être votre périphérique odb2.. mais bon … ce logiciel ci est un peut capricieux dans ces normes comme cité dans readme file. En tout cas si vous allez voir un obd usb pensez bien à prendre un avec le fameux PIcxxxx.

**Reply by Furtif** [reply]
[J’ai une ancienne dmg macOS ](https://drive.google.com/file/d/1y46RNGut0HuXhUXsYMMo69qQ0-t7Nv9L/view?usp=drive_link) mais sais pas ci ok avec votre m1 car bien la dmg était faite pour cpu intel pas pour les nouvelles m1/2 silicon Apple (ce dmg inclus déjà la v1 ecu.zip) et n’est pas la dernière version du code sur ce github, mais je vous le partage quand même.
